pkg/graph/resolvers/itgovactions/newstep/progress_to_new_step.go

```diff
@@ -0,0 +1,223 @@
+	if input.Feedback != nil {
+		errGroup.Go(func() error {
+			feedbackForRequester := &models.GovernanceRequestFeedback{
+				IntakeID:     intake.ID,
+				Feedback:     *input.Feedback,
+				SourceAction: models.GRFSAProgressToNewStep,
+				TargetForm:   models.GRFTFNoTargetProvided,
+				Type:         models.GRFTRequester,
+			}
+			feedbackForRequester.CreatedBy = adminEUAID
+
+			_, errRequesterFeedback := store.CreateGovernanceRequestFeedback(ctx, feedbackForRequester)
+			if errRequesterFeedback != nil {
+				return errRequesterFeedback
+			}
+
+			return nil
+		})
+	}
+
+	// save feedback/recommendations for GRB
+	if input.GrbRecommendations != nil {
+		errGroup.Go(func() error {
+			feedbackForGRB := &models.GovernanceRequestFeedback{
+				IntakeID:     intake.ID,
+				Feedback:     *input.GrbRecommendations,
+				SourceAction: models.GRFSAProgressToNewStep,
+				TargetForm:   models.GRFTFNoTargetProvided,
+				Type:         models.GRFTGRB,
+			}
+			feedbackForGRB.CreatedBy = adminEUAID
+
+			_, errGRBFeedback := store.CreateGovernanceRequestFeedback(ctx, feedbackForGRB)
+			if errGRBFeedback != nil {
+				return errGRBFeedback
+			}
+
+			return nil
+		})
+	}
+
+	// save admin note
+	if input.AdminNote != nil {
+		errGroup.Go(func() error {
+			adminUserInfo, errAdminUserInfo := fetchUserInfo(ctx, adminEUAID)
+			if errAdminUserInfo != nil {
+				return errAdminUserInfo
+			}
+
+			adminNote := &models.SystemIntakeNote{
+				SystemIntakeID: input.SystemIntakeID,
+				AuthorEUAID:    adminEUAID,
+				AuthorName:     null.StringFrom(adminUserInfo.CommonName),
+				Content:        null.StringFromPtr(input.AdminNote),
+			}
+
+			_, errCreateNote := store.CreateSystemIntakeNote(ctx, adminNote)
+			if errCreateNote != nil {
+				return errCreateNote
+			}
+
+			return nil
+		})
+	}
+
+	if err := errGroup.Wait(); err != nil {
+		return nil, err
+	}
+
+	return updatedIntake, nil
+}
+
+// the frontend shouldn't allow users to take any invalid actions, but validate server-side to make sure
+func isIntakeValid(intake *models.SystemIntake, newStep model.SystemIntakeStepToProgressTo) error {
+	if intake.State == models.SystemIntakeStateCLOSED {
+		return &apperrors.InvalidActionError{
+			ActionType: models.ActionTypePROGRESSTONEWSTEP,
+			Message:    "Can't take Progress to New Step action on closed intakes",
+		}
+	}
+
+	if intake.Step == models.SystemIntakeStepINITIALFORM && intake.RequestFormState == models.SIRFSNotStarted {
+		return &apperrors.InvalidActionError{
+			ActionType: models.ActionTypePROGRESSTONEWSTEP,
+			Message:    "Can't take Progress to New Step action on intakes that haven't started the Request Form",
+		}
+	}
+
+	if string(intake.Step) == string(newStep) {
+		return &apperrors.InvalidActionError{
+			ActionType: models.ActionTypePROGRESSTONEWSTEP,
+			Message:    fmt.Sprintf("Progress to New Step needs to change intake to a different step, intake is already at %v", newStep),
+		}
+	}
+
+	return nil
+}
```

Comment:
Per your comment in the PR description - I think a quick test that uses this code would be nice, but certainly not critical

While the logic in this function _is_ simple, what a test like that can help with is cases where we remove, edit, or add a step and want visibility into what logic that touches.

https://github.com/CMSgov/easi-app/blob/99869aa3c28d32e95af80e23f1ef59b5346d3056/pkg/graph/model/models_gen.go#L1044-L1049

Reply:
Tests added in https://github.com/CMSgov/easi-app/pull/2125/commits/2b9c8384bed0166d55ac119e158394f2eeef56dc.